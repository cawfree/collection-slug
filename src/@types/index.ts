export enum Network {
  ETHEREUM = 'ethereum',
  MATIC = 'matic',
  ARBITRUM = 'arbitrum',
  AVALANCHE = 'avalanche',
  OPTIMISM = 'optimism',
}
