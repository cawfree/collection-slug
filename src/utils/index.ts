export * from './fetchCollectionSlug';
export * from './fetchWithExponentialBackoff';
export * from './fetchContractAddress';
export * from './fetchSnapshotUrls';

export * from './json';
export * from './text';
export * from './winner';
