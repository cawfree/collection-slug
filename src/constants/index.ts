import Bottleneck from 'bottleneck';

import {Deployment, Network} from '../@types';

export const BASE_COLLECTION_URL = 'https://opensea.io/collection/';

export const OPENSTORE_DEPLOYMENTS: readonly Deployment[] = [
  {
    contractAddress: '0x495f947276749ce646f68ac8c248420045cb7b5e',
    network: Network.ETHEREUM,
  },
  {
    contractAddress: '0x2953399124f0cbb46d2cbacd8a89cf0599974963',
    network: Network.MATIC,
  },
];

export const DEFAULT_REDUNDANCY = 2;

export const INITIAL_BACKOFF_TIME = 5000;
export const BACKOFF_MULTIPLIER = 1.25;

export const BOTTLENECK_MAX_CONCURRENT = 4;

export const BOTTLENECK_WAYBACK_MACHINE = new Bottleneck({
  maxConcurrent: BOTTLENECK_MAX_CONCURRENT,
});

