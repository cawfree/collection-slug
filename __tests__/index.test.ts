import 'jest';

import {fetchCollectionSlug, fetchContractAddress, Network} from '../src';

jest.setTimeout(60 * 1000);

/* Please do not consider any collections referenced here as an endorsement. */

describe('collection-slug', () => {
  it('jest', expect(true).toBeTruthy);

  describe('ethereum (default)', () => {
    it('boredapeyachtclub', async () => {
      expect(await fetchCollectionSlug({
        contractAddress: '0xbc4ca0eda7647a8ab7c2061c2e118a18a936f13d',
        network: Network.ETHEREUM,
      }))
        .toEqual('boredapeyachtclub');

      expect(await fetchCollectionSlug({
        contractAddress: '0xbc4ca0eda7647a8ab7c2061c2e118a18a936f13d',
      }))
        .toEqual('boredapeyachtclub');
    });
    it('rumble-kong-league', async () => {
      expect(await fetchCollectionSlug({
        contractAddress: '0xef0182dc0574cd5874494a120750fd222fdb909a',
      }))
        .toEqual('rumble-kong-league');
    });
  });

  describe('matic', () => {
    it('layer3', async () => {
      expect(await fetchCollectionSlug({
        contractAddress: '0x200fb6e28edf0fbc9b5fabf7d39ae583981f5038',
        network: Network.MATIC,
      }))
        .toEqual('layer3');
    });
  });

  describe('arbitrum', () => {
    it('treasures-arbi', async () => {
      expect(await fetchCollectionSlug({
        contractAddress: '0xebba467ecb6b21239178033189ceae27ca12eadf',
        network: Network.ARBITRUM,
      }))
        .toEqual('treasures-arbi');
    });
  });

  describe('avalanche', () => {
    it('rich-peon-poor-peon', async () => {
      expect(await fetchCollectionSlug({
        contractAddress: '0x4c5a8b71330d751bf995472f3ab8ceb06a98dd47',
        network: Network.AVALANCHE,
      }))
        .toEqual('rich-peon-poor-peon');
    });
  });

  describe('optimism', () => {
    it('optimism-quests', async () => {
      expect(await fetchCollectionSlug({
        contractAddress: '0xfa14e1157f35e1dad95dc3f822a9d18c40e360e2',
        network: Network.OPTIMISM,
      }))
        .toEqual('optimism-quests');
    });
  });

  describe('openstore', () => {

    it('missing-token-id', async () => {
      await expect(
        fetchCollectionSlug({
          contractAddress: '0x2953399124f0cbb46d2cbacd8a89cf0599974963',
          network: Network.MATIC,
        })
      ).rejects.toMatchSnapshot();
    });

    describe('matic', () => {

      it('artisantspecial', async () => {
        await expect(
          fetchCollectionSlug({
            contractAddress: '0x2953399124f0cbb46d2cbacd8a89cf0599974963',
            network: Network.MATIC,
            tokenId: '113248483929477389204957219413891167927065839510717550429608723920230394365404',
          }),
        ).resolves.toBe('artisantspecial');
      });

      it('wandievents', async () => {
        await expect(
          fetchCollectionSlug({
            contractAddress: '0x2953399124f0cbb46d2cbacd8a89cf0599974963',
            network: Network.MATIC,
            tokenId: '94149156980946697317614428571944365609665181621202038530767030033582598931348',
            }),
        ).resolves.toBe('wandievents');
      });
    });
  });

  describe('fetchContractAddress', () => {
    it('boredapeyachtclub', async () => {
      expect(
        await fetchContractAddress({
          collectionSlug: 'boredapeyachtclub',
        })
      ).toBe('0xbc4ca0eda7647a8ab7c2061c2e118a18a936f13d');
    });
    it('rumble-kong-league', async () => {
      expect(
        await fetchContractAddress({
          collectionSlug: 'rumble-kong-league',
        }),
      ).toBe('0xef0182dc0574cd5874494a120750fd222fdb909a');
    });
    it('renftlabs', async () => {
      expect(
        await fetchContractAddress({
          collectionSlug: 'renftlabs',
        }),
      ).toBe('0x0db8c099b426677f575d512874d45a767e9acc3c');
    });
  });
});

