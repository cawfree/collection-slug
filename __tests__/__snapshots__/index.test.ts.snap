// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`collection-slug openstore missing-token-id 1`] = `[Error: To find the collectionSlug on the OPENSTORE contract, you must specify a tokenId.]`;
